test/functional.jl

```diff
@@ -239,3 +239,33 @@ end
 @test_throws DimensionMismatch Base.IteratorND(1:2, (2,3))
 
 @test collect(Base.Generator(+, [1,2], [10,20])) == [11,22]
+
+# partition(c, n)
+
+let v = collect(Base.partition([1,2,3,4,5], 1))
+    @test all(i->v[i][1] == i, v)
+end
+
+let v = collect(Base.partition([1,2,3,4,5], 2))
+    @test v[1] == [1,2]
+    @test v[2] == [3,4]
+    @test v[3] == [5]
+end
+
+let v = collect(Base.partition(enumerate([1,2,3,4,5]), 3))
+    @test v[1] == [(1,1),(2,2),(3,3)]
+    @test v[2] == [(4,4),(5,5)]
+end
+
+for n in [5,6]
+    @test collect(Base.partition([1,2,3,4,5], n))[1] == [1,2,3,4,5]
+    @test collect(Base.partition(enumerate([1,2,3,4,5]), n))[1] ==
+          [(1,1),(2,2),(3,3),(4,4),(5,5)]
+end
+
+
+@test_throws AssertionError Base.partition([1,2,3,4,5], 0)
+@test_throws AssertionError Base.partition([1,2,3,4,5], -1)
+
+@test join(map(x->string(x...), Base.partition("Hello World!", 5)), "|") ==
```

Comment:
OK, great! But then please add non-ASCII characters to the string so that any regression would be caught.